### PR TITLE
[fixes #502][Fixes #476] Fixes auto-scrolling of FinPointFigure bounds.

### LINKED ISSUE #476: FreeformFinConfig: On initial open, auto-scroll should center to fin bounds.
> The fin editor always opens up zoomed out to the parent component.  
1. the zoom bounds (size)  don't make sense. 
    - [ ] scroll bars not present when they should be  (canvas size is too small?)
    - [ ] bounds are too small for the fin itself.  (should be mount + fins)
2. The zoom bounds (origin) seems to start at the beginning of the fin (which is probably the desired behavior)

### LINKED ISSUE #502: Side view horizontal centering problems
No ORK file is attached to this one, just follow along to reproduce.  Screenshots show 15.03 on left, new alpha 10 on right.

Start with a new design.  Add a default body tube:
centering1

So far so good.  Now add to the body tube a default freeform fin set.  In the new build, set the number of fins to 3.  Things are starting to go awry:
centering2

Now edit the shape of the fins (do the same change to each).  Between the first and second points, add a new point at (-20, 1).  In the new build, having all that extra fin area way to the left somehow pulls the side view even more to the left,  Old build handles it properly.
centering3

As an aside, neither the old nor new build knows how to zoom this properly in the 3D view:
centering4

Back to side view.  For the final step, delete the fin set, leaving only the body tube.  15.03 returns to the original view, while alpha 10 holds the same position as before:
centering5

### PR DESCRIPTION
**Pull request by teyrana**
1. Simplifies code in FlightConfiguration
2. Adds Interface to indicate implementation of getInstanceBoundingBox function
3. Adjusts test to ensure that refactoring still behaves as desired
4. Adds a fix to the Intersection testing code in FreeformFinSet.setPoint

@neilweinstock
This fixes issue #476, as reported.

### UI Changes:

Here's the new zoom results:
results

### Jar for testing:
https://drive.google.com/file/d/1rrOCsJG-2JZ9C_G5fI8uY3GvHrVkDW7R/view?usp=sharing

### CODE REVIEW

**Comment by neilweinstock** [comment]
Looks great int the screenshot, can't wait to try it out.

**Comment by teyrana** [comment]
@neilweinstock 
You saw the jar link above, right?

**Comment by neilweinstock** [comment]
No I did not!  Sorry about that, testing now.

**Comment by neilweinstock** [comment]
OK, just tried it.

For me, this is only a partial fix.

The good: it fixes the problem where the entire fin was not on-screen at the start.

The bad: it still seems determined to show the complete parent object in the view.  This is not IMHO useful behavior.  Example:
image
This forces me to do a bunch of zooming and scrolling before I can reasonably start editing.  I want the initial zoom to be bounded by the fin, and I honestly don't care about the bounds of the parent object.  E.g.:
image

That would enable me to start editing immediately.  Note that 15.03 works this way, although it is a bit buggy (sometimes doesn't show the entire fin, but it's usually close).

I mentioned this in the original issue (#472) but it might have gotten a little lost among the other things discussed there.

File: core/src/net/sf/openrocket/rocketcomponent/Rocket.java [hunk 1]
```diff
@@ -2,17 +2,14 @@
...
 import org.slf4j.LoggerFactory;
 
 import net.sf.openrocket.l10n.Translator;
 import net.sf.openrocket.rocketcomponent.position.AxialMethod;
 import net.sf.openrocket.startup.Application;
-import net.sf.openrocket.util.ArrayList;
-import net.sf.openrocket.util.Coordinate;
-import net.sf.openrocket.util.MathUtil;
-import net.sf.openrocket.util.StateChangeListener;
-import net.sf.openrocket.util.UniqueID;
```
Comment (hola-holo) [inline core/src/net/sf/openrocket/rocketcomponent/Rocket.java:13]: our static analysis engine detects that your pull request includes a change from an explicit import to a wildcard import. According to the best practice (http://www.javadude.com/articles/importondemandisevil.html), import star (import on demand) is a bad practice, wildcard import can sometimes cause name conflicts and ambiguities.

**Comment by teyrana** [comment]
@neilweinstock 
Yea, I see what you mean.  I'll circle back and adjust the auto-zoom bounds.

**Comment by neilweinstock** [comment]
Thanks!

**Comment by teyrana** [comment]
Okay, that was a lot more complicated than expected.  It turns out the figures didn't really understand auto-zooming to anything except their full bounds.  So this was more of a new feature than a fix, exactly.  But it should do what you want, now.

**Comment by neilweinstock** [comment]
Daniel,
Sorry for my slowness getting back on this one.

It's closer but not quite there yet.  It seems as if the zoom level is being set correctly, but not the scroll position.

Consider the following rocket (ORK also attached):
image

Starting with the main (middle) fins: opening up the editor looks good:
image

Now one of the front fins:
image
It has set the scroll position to the back of the parent body tube, and so the fin is off the left side of the window.  If I scroll it into view, you can see the that zoom is perfect:
image

I'm not sure why the fin is jammed into the upper half of the edit window but that's minor.

Now let's try one of the rear fins that's mounted to the tail cone:
image

This time the vertical scroll is off.  If I adjust it, the whole fine does (barely) fit in the window:
image

Here's the ORK file.
[blackfish_golden_sunset.ork.zip](https://github.com/openrocket/openrocket/files/4878978/blackfish_golden_sunset.ork.zip)

- Neil

**Comment by teyrana** [comment]
@neilweinstock 
[Okay, third time's the charm? ](https://drive.google.com/file/d/1edHNbPGd4wP_O7x_U29O8zG751md8xUv/view?usp=sharing)

I've made a few changes, which afaict fix the auto-zooming issues.

(code is not yet ready for review)

**Comment by neilweinstock** [comment]
Looks good!

**Pull request merged**